src/platform.rs

```diff
@@ -351,6 +391,20 @@ pub fn sse41_detected() -> bool {
     false
 }
 
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+#[inline(always)]
+pub fn sse2_detected() -> bool {
+    // A testing-only short-circuit.
+    if cfg!(feature = "no_sse2") {
+        return false;
+    }
+    // Static check, e.g. for building with target-cpu=native.
+    #[cfg(target_feature = "sse2")]
+    {
+        return true;
+    }
```

Comment:
The unconditional return was missing at the end here. Rustc almost always assumes that SSE2 is supported, but you can convince it not to and trigger a build break like this:

    cargo rustc --target i686-unknown-linux-musl -- -C target-cpu=i386

I've gone ahead and fixed this in https://github.com/BLAKE3-team/BLAKE3/pull/116.

Reply:
> Rustc almost always assumes that SSE2 is supported, but you can convince it not to and trigger a build break like this:

Okay, that's what I suspected.

> I've gone ahead and fixed this in #116.

Thanks. I had something like that in my original version, but I wasn't sure how to get the tests to pass with the compiler warning that this part was unreachable.